### DISCUSSION TITLE
Enabling `--allow-local-files` for the marp slide generation in the `vscode` extension

### DISCUSSION THREAD

**Discussion by spkane** [marp-team/marp#449] [Q&A] [answered]
Is there any preference or other technique to enable the  `--allow-local-files` option in a trusted workspace, so that I can render my slides properly using local images that are not accessible via a URL?

**Comment by yhatt** [comment]
Normally Marp for VS Code always enable `--allow-local-files` option while exporting via Marp CLI.

If you enabled an experimental option `markdown.marp.strictPathResolutionDuringExport` or run export command over the remote, `--allow-local-files` may disable. Check if the option is disabled in VS Code preference.

**Comment by spkane** [comment]
I still cannot see my CSS theme background image when using vscode to build the PDF. It is likely something environmental, but I would love to get a bit more debugging info to determine if I need to tweak it in my setup or if it is a bug.

Is there an easy way to see the actual command running and output generated during the export and the working directory? The debug console appears empty.

**Reply by yhatt** [reply]
Open VS Code's [DevTools](https://developer.chrome.com/docs/devtools/) by [Help] -> [Toggle Developer Tools], and check the log in [Console] tab.

**Reply by spkane** [reply, marked as answer]
I spent some time with this today and after about 90 minutes I was able to get the custom CSS working correctly inside VScode. The Dev Tools weren't as useful as I would wish, simply because it was hard to tell what marp was using as the working directory, what config was being passed to marp, etc.

That being said, I was able to refactor my directory layout a bit, enable `Strict Path Resolution During Export` and then add the relative paths to my two custom CSS files to the `Themes` setting and then rendering is now working properly in VScode as well as from the CLI.

**Comment by spkane** [comment]
@yhatt Thank you for the hints. They helped me get this resolved.
